feat: Track separate event for auction screen view GRO-363

jonallured opened the pull request:
This is likely the wrong way to do this because it's not using Cohesion, but what I'm trying to match is this event in Force:

https://github.com/artsy/force/blob/master/src/lib/analytics/trackPageView.ts#L52

https://artsyproduct.atlassian.net/browse/GRO-363

/cc @artsy/grow-devs

ArtsyOpenSource commented:
<!--
  0 failure: 
  1 warning:  ❌ **An error occu...
  
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 51d60f2c0ec33b8b5b874e54f6cf8b586ed01da5
</p>

pvinis commented:
for screens, we usually do something like https://github.com/artsy/eigen/blob/fe5163bdc0a6d0d7923f061ca516e08b87848862/src/lib/Scenes/MyCollection/MyCollection.tsx#L70-L75.

if it's a rogue track event that we just want to match, then cool to add it just like that and make a ticket to add it to cohesion and do it right.

MounirDhahri commented on src/lib/Scenes/Sale/Components/SaleHeader.tsx:
```diff
@@ -19,6 +20,12 @@ interface Props {
   const saleTimeDetails = saleTime(sale)
+  const { trackEvent } = useTracking()
+
+  trackEvent({
+    auction_slug: sale.slug,
+    name: "Auctions Pageview",
+  })
```
**thought:** I think you better track this in `Sale.tsx` here https://github.com/artsy/eigen/blob/7d6e9ca3eecce96eaf4534b07aa53bf56cdf8d78/src/lib/Scenes/Sale/Sale.tsx#L332 and add the tracking event as in here https://github.com/artsy/eigen/blob/7d6e9ca3eecce96eaf4534b07aa53bf56cdf8d78/src/lib/Scenes/Sale/Sale.tsx#L231-L239

If for some reason, it has to be inside the `SaleHeader` then it would need to be wrapped with `useEffect` to avoid triggering it on screen re-rerenders
```suggestion
useEffect(() => {
  trackEvent({
    auction_slug: sale.slug,
    name: "Auctions Pageview",
  })
},[])
  
```

jonallured replied:
Oh good call @MounirDhahri !! 🏆

jonallured commented:
Ok I sorta started over based on feedback from @MounirDhahri and moved the track call with b7ecdfe. So now I'm doing the call inside the query renderer wrapped with a `useEffect` so I think this is better but lmk if I missed anything! Oh I also moved the hard-coded bits into that `tracks` object to follow the existing pattern in that file.

jonallured commented on src/lib/Scenes/Sale/Sale.tsx:
```diff
@@ -328,6 +334,12 @@ export const SaleQueryRenderer: React.FC<{ saleID: string; environment?: RelayMo
+  const { trackEvent } = useTracking()
+
+  useEffect(() => {
+    trackEvent(tracks.pageView(saleID))
```
The saleID here is a slug, right? 🤔

MounirDhahri replied:
Yes 🙌

MounirDhahri commented:
Thanks for making the changes 🙌

jonallured commented:
Rebased this on latest master and it's green so going to merge for now.
